src/makeCalculator.js

```diff
@@ -37,7 +37,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add: (x, y) => {
+      return x + y;
+    },
+    subtract: (x, y) => {
+      return x - y;
+    },
+    multiply: (x, y) => {
+      return x * y;
+    },
+    divide: (x, y) => {
+      return x / y;
+    },
+    reset() {
+      calculator.result = 0;
+
+      return this;
+    },
+    operate(operation, number) {
+      calculator.result = operation(calculator.result, number);
```

Comment:
Smart. Also, you could try to use `call`,`apply` or `bind` method, if you would like to call operation function without `calculator` reference